zio-config/src/main/scala/zio/config/package.scala

```diff
@@ -16,16 +16,21 @@ package object config extends Sources {
   def config[A]: ZIO[Config[A], Nothing, A] =
     ZIO.accessM(_.config.config)
 
-  def getConfigValue(path: String): ZIO[ConfigSource, ReadError, String] =
-    ZIO.accessM(_.configService.getConfigValue(path))
+  def getConfigValue(
+    path: String
+  ): ZIO[ConfigSource[String, String], ReadError[String, String], PropertyTree[String, String]] =
+    ZIO.accessM(_.configSourceService.getConfigValue(path))
 
-  type ReadErrors = ::[ReadError]
+  type ReadErrors[K, V] = ::[ReadError[K, V]]
 
   object ReadErrors {
-    def apply(a: ReadError, as: ReadError*): ReadErrors =
+    def apply[K, V](a: ReadError[K, V], as: ReadError[K, V]*): ReadErrors[K, V] =
       ::(a, as.toList)
 
-    def concat(l: ReadErrors, r: ReadErrors): ReadErrors =
+    def concat[K, V](l: ReadErrors[K, V], r: ReadErrors[K, V]): ReadErrors[K, V] =
       ::(l.head, l.tail ++ r)
+
+    def asThrowable[K, V, A](readErrors: ReadErrors[K, V]): RIO[system.System, A] =
+      zio.system.lineSeparator.flatMap(t => ZIO.fail(new RuntimeException(readErrors.mkString(t))))
```

Comment:
A helper for the user in case they want to convert to throwable

Reply:
How do you feel about extending the error types by `Exception`?

```
sealed trait ReadError extends Exception {
  val key: String
}

  final case class ReadErrors(errors: ::[ReadError]) extends Exception {
    override def getMessage(): String = ???
  }

  object ReadErrors {
    def apply(a: ReadError, as: ReadError*): ReadErrors =
      ReadErrors(::(a, as.toList))

    def concat(l: ReadErrors, r: ReadErrors): ReadErrors =
      ReadErrors(::(l.errors.head, l.errors.tail ++ r.errors))
  }
```

Reply:
I prefer errors not extending exception unless it really benefits us. Especially in this case, bootstrapping a config happens at the main resulting in some read errors. And we usually resolve it to folding over the result and logging the error  and zio.succeed(1)